src/library/scala/concurrent/duration/Duration.scala

```diff
@@ -124,6 +124,8 @@ object Duration {
       Undefined
     else if (nanos > Long.MaxValue || nanos < Long.MinValue)
       throw new IllegalArgumentException("trying to construct too large duration with " + nanos + "ns")
+    else if (nanos < 0)
+      fromNanos((nanos - 0.5).toLong)
```

Comment:
@Blaisorblade My first attempt was trying to stay in line with the existing code which I assumed had a reason not to `round`. But I think you're right, especially if there are bugs in `(double+0.5).toLong`, that `round` is the only reasonable option here.